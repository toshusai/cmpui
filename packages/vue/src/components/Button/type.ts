export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
