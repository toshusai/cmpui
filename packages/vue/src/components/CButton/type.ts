export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
