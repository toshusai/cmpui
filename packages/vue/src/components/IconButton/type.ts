export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
