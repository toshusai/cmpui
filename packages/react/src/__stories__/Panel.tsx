export const Panel = (
  props: {
    color: string;
    name: string;
  } & React.HTMLAttributes<HTMLDivElement>,
) => {
  return (
    <div
      style={{
        background: props.color,
        width: "100%",
        height: "100%",
        display: "flex",
        overflow: "auto",
        position: "relative",
      }}
      {...props}
    >
      <div
        style={{
          width: "100%",
          pointerEvents: "none",
          height: "100%",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          color: "white",
          fontSize: "32px",
          position: "absolute",
        }}
      >
        {props.name}
      </div>
      {/* <pre
        style={{
          fontSize: "12px",
          fontFamily: "monospace",
          margin: 0,
          padding: 8,
        }}
      >
        <code>{JSON.stringify(node?.node.toJson(), null, 4)}</code>
      </pre> */}
    </div>
  );
};
