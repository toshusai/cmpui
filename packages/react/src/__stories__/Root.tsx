export const Root = (props: { children: React.ReactNode }) => {
  return (
    <div
      style={{
        height: "512px",
        width: "100vw",
        display: "flex",
        padding: "4px",
        boxSizing: "border-box",
      }}
    >
      {props.children}
    </div>
  );
};
