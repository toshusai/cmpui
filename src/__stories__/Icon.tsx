export function Icon({
  name,
  filled,
  style,
  size,
  ...props
}: {
  name: string;
  filled?: boolean;
  size?: number;
} & React.HTMLAttributes<HTMLSpanElement>) {
  return (
    <span
      className="material-symbols-outlined"
      data-filled={filled}
      {...props}
      style={{
        fontSize: size,
        ...style,
      }}
    >
      {name}
    </span>
  );
}
