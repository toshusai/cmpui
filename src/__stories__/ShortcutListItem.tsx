import { ListItem } from "..";
import "./stories.css";

export function ShortcutListItem({
  disabled,
  shortcut,
  as = ListItem,
  children,
  ...props
}: {
  children: React.ReactNode;
  shortcut?: string;
  disabled?: boolean;
  as?: React.ElementType;
} & React.ComponentProps<typeof ListItem>) {
  const Component = as;
  return (
    <Component disabled={disabled} className="shortcut-list-item" {...props}>
      <div style={{ width: "100%" }}>{children}</div>
      {shortcut && (
        <div
          style={{
            fontSize: "10px",
            marginLeft: 16,
            whiteSpace: "nowrap",
          }}
        >
          {shortcut}
        </div>
      )}
    </Component>
  );
}
