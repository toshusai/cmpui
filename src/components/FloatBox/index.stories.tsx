import type { Meta, StoryObj } from "@storybook/react";
import { FloatBox } from ".";
import { ShortcutListItem } from "../../__stories__/ShortcutListItem";

const meta: Meta<typeof FloatBox> = {
  title: "Components/Float/FloatBox",
  component: FloatBox,
};

export default meta;
type Story = StoryObj<typeof FloatBox>;

export const Basic: Story = {
  render: function Render() {
    return (
      <FloatBox
        style={{
          width: 128,
        }}
      >
        <ShortcutListItem rounded size="S" shortcut="⌘ Z">
          Undo
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⇧ ⌘ Z">
          Redo
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ X">
          Cut
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ C">
          Copy
        </ShortcutListItem>
        <ShortcutListItem rounded size="S" shortcut="⌘ V">
          Paste
        </ShortcutListItem>
      </FloatBox>
    );
  },
};
