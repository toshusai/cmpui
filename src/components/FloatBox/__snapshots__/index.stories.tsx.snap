// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Float/FloatBox Basic smoke-test 1`] = `
<div class="cmpui_float-box__root"
     style="width: 128px;"
>
  <li class="cmpui_list-item__root shortcut-list-item"
      data-size="S"
      data-rounded="true"
  >
    <div style="width: 100%;">
      Undo
    </div>
    <div style="font-size: 10px; margin-left: 16px; white-space: nowrap;">
      ⌘ Z
    </div>
  </li>
  <li class="cmpui_list-item__root shortcut-list-item"
      data-size="S"
      data-rounded="true"
  >
    <div style="width: 100%;">
      Redo
    </div>
    <div style="font-size: 10px; margin-left: 16px; white-space: nowrap;">
      ⇧ ⌘ Z
    </div>
  </li>
  <li class="cmpui_list-item__root shortcut-list-item"
      data-size="S"
      data-rounded="true"
  >
    <div style="width: 100%;">
      Cut
    </div>
    <div style="font-size: 10px; margin-left: 16px; white-space: nowrap;">
      ⌘ X
    </div>
  </li>
  <li class="cmpui_list-item__root shortcut-list-item"
      data-size="S"
      data-rounded="true"
  >
    <div style="width: 100%;">
      Copy
    </div>
    <div style="font-size: 10px; margin-left: 16px; white-space: nowrap;">
      ⌘ C
    </div>
  </li>
  <li class="cmpui_list-item__root shortcut-list-item"
      data-size="S"
      data-rounded="true"
  >
    <div style="width: 100%;">
      Paste
    </div>
    <div style="font-size: 10px; margin-left: 16px; white-space: nowrap;">
      ⌘ V
    </div>
  </li>
</div>
`;
